Misc fixes (#1619)
dotnet/csharpstandard: standard/unsafe-code.md

In an unsafe context, a *type* ([§8.1](types.md#81-general)) can be a *pointer_type* as well as a *value_type*, a *reference_type*, or a *type_parameter*. In an unsafe context a *pointer_type* may also be the element type of an array ([§17](arrays.md#17-arrays)). A *pointer_type* may also be used in a typeof expression ([§12.8.18](expressions.md#12818-the-typeof-operator)) outside of an unsafe context (as such usage is not unsafe).

> A *pointer_type* is written as a *value_type* that is an unmanaged type ([§8.8](types.md#88-unmanaged-types)) or the keyword `void`, followed by a `*` token:

Reviewer: ```suggestion
A *pointer_type* is written as a *value_type* that is an unmanaged type ([§8.8](types.md#88-unmanaged-types)) or the keyword `void`, followed by one or more `*` tokens:
```

A *pointer_type* is written as an *unmanaged_type* ([§8.8](types.md#88-unmanaged-types)) or the keyword `void`, followed by a `*` token:

```ANTLR

> pointer_type

Author: ```suggestion
```

A *pointer_type* is written as an *unmanaged_type* ([§8.8](types.md#88-unmanaged-types)) or the keyword `void`, followed by a `*` token:

```ANTLR
pointer_type
A *pointer_type* is written as a *value_type* that is an unmanaged type ([§8.8](types.md#88-unmanaged-types)) or the keyword `void`, followed by a `*` token:

```ANTLR
pointer_type
    : value_type ('*')+

>     | pointer_type('*')

Author: ```suggestion
```

```ANTLR
pointer_type

>     : unmanaged_type ('*')+

Author: This change does introduce a left recursion in the grammar. In 8.8, an *unmanaged_type* is defined in the grammar as a *value_type* or a *pointer_type*. Then, in the prose the *value_type* is constrained to the builtin value types, enum types, and struct types whose members are only other unmanaged types in that list (paraphrasing here in the comment).

It's true that a *pointer_type* can only refer to an *unmanaged_type*, as written in the prose in line 120. But, to avoid the recursion, a bit fix might be to revert this change and expand the prose on line 120.

Reviewer: cc @Nigel-Ecma for thoughts on the impact of introducing left recursion.

>     : unmanaged_type ('*')+

Author: Revert grammar and clarify rule:

```suggestion
A *pointer_type* is written as a *value_type* that is an unmanaged type ([§8.8](types.md#88-unmanaged-types)) or the keyword `void`, followed by a `*` token:

```ANTLR
pointer_type
    : value_type ('*')+
    | pointer_type('*')
```
